src/de.rs

```diff
@@ -304,43 +305,29 @@ impl<'de> de::IntoDeserializer<'de, DeserializerError> for Value {
     }
 }
 
-struct MapVisitor<I> {
-    iter: I,
-    value: Option<Value>,
+pub struct ValueDeserializer<E> {
+    value: Value,
+    error: PhantomData<E>,
```

Comment:
```suggestion
    error: PhantomData<fn() -> E>,
```

I'd imagine the variance difference doesn't really matter here but would this be more appropriate?